pkg/protocol/engine/congestioncontrol/scheduler/drr/scheduler.go

```diff
@@ -244,6 +244,11 @@ func (s *Scheduler) AddBlock(block *blocks.Block) {
 	}
 }
 
+// Reset resets the component to a clean state as if it was created at the last commitment.
+func (s *Scheduler) Reset() {
+	// TODO: clear all buffers? currently irrelevant because the scheduler is not relevant for commitments.
```

Comment:
But it's relevant for tip selection, because tips are added to the tip manager only after being scheduled. So I think we should clear all buffers first, and only then reset the TipManager. Otherwise it might happen that some old blocks will be added as tips even after resetting everything.

Reply:
I implemented the reset strategy together with Andrew